internal/controllers/machinedeployment/machinedeployment_rollout_planner_test.go

```diff
@@ -535,15 +598,93 @@ func machineSetControllerMutator(log *fileLogger, ms *clusterv1.MachineSet, scop
 		}
 	}
 
-	// if too many replicas, delete exceeding machines.
-	// exceeding machines are deleted in predictable order, so it is easier to write test case and validate rollout sequences.
+	// if too many replicas, delete or move exceeding machines.
+	// exceeding machines are deleted or moved in predictable order, so it is easier to write test case and validate rollout sequences.
 	// e.g. if a ms has m1,m2,m3 created in this order, m1 will be deleted first, then m2 and finally m3.
-	machinesToDelete := max(ptr.Deref(ms.Status.Replicas, 0)-ptr.Deref(ms.Spec.Replicas, 0), 0)
+	// Note: replicas still pending AcknowledgeMove should not be counted when computing the numbers of machines to delete, because those machines are not included in ms.Spec.Replicas yet.
```

Comment:
But are they included in ms.Status.Replicas?

How will the prod controller behave?

Reply:
I have added my note to align fake machine controller to the implementation in prod code to the tracking issue for better visibility